source/common/http/conn_manager_utility.cc

```diff
@@ -131,6 +131,70 @@ void ConnectionManagerUtility::mutateRequestHeaders(Http::HeaderMap& request_hea
   if (config.tracingConfig()) {
     Tracing::HttpTracerUtility::mutateHeaders(request_headers, runtime);
   }
+  mutateXfccRequestHeader(request_headers, connection, config);
+}
+
+void ConnectionManagerUtility::mutateXfccRequestHeader(Http::HeaderMap& request_headers,
+                                                       Network::Connection& connection,
+                                                       ConnectionManagerConfig& config) {
+  if (config.forwardClientCert() != Http::ForwardClientCertType::AlwaysForwardOnly &&
+      !(connection.ssl() && connection.ssl()->peerCertificatePresented())) {
+    request_headers.removeForwardedClientCert();
+    return;
+  }
+  std::string clientCertDetails;
```

Comment:
I meant you can directly use append(), since you're only doing `+=` for string, i.e. something similar to here:
https://github.com/lyft/envoy/blob/master/source/common/grpc/common.cc#L137

Instead of even building a std::string. That will avoid to copy that when you append at the end. It is OK to optimize that based on perf analysis.